test/parallel/test-worker-message-port-onmessage-stop.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+const common = require('../common');
+const tick = require('../common/tick');
+const assert = require('assert');
+const { MessageChannel } = require('worker_threads');
+
+// Make sure that messages start being received when .onmessage is first set,
+// but when it is unset, messages end up being discarded.
+// Refs: https://github.com/nodejs/node/issues/26463
+
+{
+  const { port1, port2 } = new MessageChannel();
+  port1.onmessage = common.mustCall(({ data }) => {
+    assert.strictEqual(data, 'hello');
+    port1.onmessage = null;
+    port2.postMessage('world');
+
+    // Give some event loop ticks for the 'world' message to be discarded.
+    tick(10, () => {
```

Comment:
I’m not sure I understand what you are saying – what would wrongly activating (= calling?) `.onmessage` mean?

Reply:
`port1.onmessage = null;`  // port1 is ditached from messages

`port2.postMessage('world');`

What we want to check is `port1.onmessage` should not be called henceforth. But if it is being called by error, we don't have a means to catch it?

Reply:
@gireeshpunathil How do you test that something that is not a function isn’t called?

We can only check that there’s nothing in the queue after a few ticks, and that’s what setting `onmessage = common.mustNotCall()` is there for

Reply:
@addaleax - I see your point. What I thought was something like this:
```js
  const { port1, port2 } = new MessageChannel();
  port1.onmessage = common.mustCall(({ data }) => {
    assert.strictEqual(data, 'hello');
    port1.onmessage = null;
    port2.postMessage('world');
  }, 1);
  port2.postMessage('hello');
```

Given that we are dealing with only one thread here, and the port1 being detached from its message channel would happen synchronously, we should see the result immediately, and the subsequent emit from port2 itself could be subjected for the assert test? assert that we don't call the handler more than once?

Reply:
> assert that we don't call the handler more than once?

Still not quite sure – we do use `common.mustCall()` currently for the “outermost” `onmessage`? Can you provide an example of changes to the test that you would like to see?

Reply:
This snippet itself is what I am proposing. if the intent of the test is to: `Make sure that messages start being received when .onmessage is first set, but when it is unset, messages end up being discarded.` , then won't this sequence test it reliably?

Once the `onmessage` is unset, the only possible error that can occur is it is bound to the function somehow, and the callback (outer onmessage, in fact now the only one in the program) will be invoked, and that will be captured by the common.mustCall that mandates only one invocation?

```js
const { port1, port2 } = new MessageChannel();
port1.onmessage = common.mustCall(({ data }) => {
  assert.strictEqual(data, 'hello');
  port1.onmessage = null;
  port2.postMessage('world');
}, 1);
port2.postMessage('hello');
```

Reply:
@gireeshpunathil There is already a `common.mustCall()` in the test in this PR for the outer onmessage, and another error condition which we want to check for would be that the second message is just queued up and not discarded (as it would be before this patch), which wouldn’t be caught if we leave `onmessage` as `null` until the test ends?

Reply:
ah, ok! I realize it now, thanks.